openmc/region.py

```diff
@@ -29,6 +29,17 @@ def __invert__(self):
     def __str__(self):
         return ''
 
+    def __eq__(self, other):
+        if not isinstance(other, type(self)):
+            return False
+        elif str(self) != str(other):
```

Comment:
We should probably add `__repr__` and `__hash__` methods to subclasses of Region and Surface so that this can be handled a bit better -- I can take care of that in a separate PR though.